packages/keyring-eth-ledger-bridge/src/ledger-error-handler.test.ts

```diff
@@ -0,0 +1,200 @@
+      const originalMessage = 'Unknown transport error';
+      const error = createTransportStatusError(
+        originalMessage,
+        unknownStatusCode,
+      );
+
+      expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+        LedgerStatusError,
+      );
+
+      let thrownError: unknown;
+      try {
+        handleLedgerTransportError(error, fallbackMessage);
+      } catch (error_: unknown) {
+        thrownError = error_;
+      }
+      expect(thrownError).toBeInstanceOf(LedgerStatusError);
+      expect((thrownError as LedgerStatusError).statusCode).toBe(
+        unknownStatusCode,
+      );
+      expect((thrownError as LedgerStatusError).message).toBe(originalMessage);
+    });
+  });
+
+  describe('when error is not TransportStatusError', () => {
+    it('should re-throw Error instances as-is', () => {
+      const originalError = new Error('Original error message');
+
+      expect(() =>
+        handleLedgerTransportError(originalError, fallbackMessage),
+      ).toThrow(originalError);
+
+      expect(() =>
+        handleLedgerTransportError(originalError, fallbackMessage),
+      ).toThrow('Original error message');
+    });
+
+    it('should create new Error with fallback message for non-Error instances', () => {
+      const nonErrorValue = 'string error';
+
+      const throwingFunction = (): never =>
+        handleLedgerTransportError(nonErrorValue, fallbackMessage);
+
+      expect(throwingFunction).toThrow(Error);
+      expect(throwingFunction).toThrow(fallbackMessage);
+    });
+
+    it('should create new Error with fallback message for null', () => {
+      const throwingFunctionNull = (): never =>
+        handleLedgerTransportError(null, fallbackMessage);
+
+      expect(throwingFunctionNull).toThrow(Error);
+      expect(throwingFunctionNull).toThrow(fallbackMessage);
+    });
+
+    it('should create new Error with fallback message for undefined', () => {
+      const throwingFunctionUndefined = (): never =>
+        handleLedgerTransportError(undefined, fallbackMessage);
+
+      expect(throwingFunctionUndefined).toThrow(Error);
+      expect(throwingFunctionUndefined).toThrow(fallbackMessage);
+    });
+
+    it('should create new Error with fallback message for objects without Error prototype', () => {
+      const plainObject = { message: 'not an error' };
+
+      const throwingFunction = (): never =>
+        handleLedgerTransportError(plainObject, fallbackMessage);
+
+      expect(throwingFunction).toThrow(Error);
+      expect(throwingFunction).toThrow(fallbackMessage);
+    });
+  });
+
+  describe('return type', () => {
+    it('should have never return type (always throws)', () => {
+      const error = new Error('test');
+
+      // This test verifies that the function always throws
+      // by ensuring it never returns normally
+      expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+        'test',
+      );
+    });
```

Comment:
Looks very similar to the other test with `originalError`.

We could do it that way (but that's gonna be a compile time error in case the signature changes):

```suggestion
    it('should have never return type (always throws)', () => {
      type ReturnTypeIsNever<Function> = ReturnType<Function> extends never ? true : false;
      
      const isNever: ReturnTypeIsNever<typeof handleLedgerTransportError> = true;
      expect(isNever).toBe(true);
```

WDYT?

Other alternative would be to use `tsd` for testing type (we do this in some other packages in that monorepo)